crates/iceberg-catalog/src/service/storage/mod.rs

```diff
@@ -118,34 +119,39 @@ impl StorageProfile {
         }
     }
 
-    #[must_use]
-    pub fn initial_tabular_location(
-        &self,
-        namespace_id: NamespaceIdentUuid,
-        table_id: TabularIdentUuid,
-    ) -> String {
+    /// Get the base location of this Storage Profiles
+    ///
+    /// # Errors
+    /// Can fail for un-normalized profiles.
+    pub fn base_location(&self) -> Result<Location, ValidationError> {
         match self {
-            StorageProfile::S3(profile) => profile.tabular_location(namespace_id, table_id),
-            StorageProfile::Azdls(profile) => {
-                profile.initial_tabular_location(namespace_id, table_id)
-            }
+            StorageProfile::S3(profile) => profile.base_location().map(Into::into),
+            StorageProfile::Azdls(profile) => profile.base_location(),
             #[cfg(test)]
-            StorageProfile::Test(_) => format!("/tmp/{namespace_id}/{table_id}"),
+            StorageProfile::Test(_) => std::str::FromStr::from_str("file://tmp/").map_err(|_| {
```

Comment:
Location::from_str(...)?

Reply:
I had that first - problem is that FromStr is not `use`ed by default, leading to a conditional use which I like even less.